src/hotspot/share/oops/klassVtable.cpp

```diff
@@ -1571,7 +1571,7 @@ void klassVtable::verify(outputStream* st, bool forced) {
   // verify consistency with superKlass vtable
   Klass* super = _klass->super();
   if (super != nullptr) {
-    InstanceKlass* sk = InstanceKlass::cast(super);
+    InstanceKlass* sk = InstanceKlass::cast(super); // why are we sure this is InstanceKlass??
```

Comment:
I don't think it is guaranteed to be an IK. But AFAICS we don't actually need an IK here anyway, we should be able to use `super` directly.